templates/tower-project.yaml

```diff
@@ -0,0 +1,365 @@
+
+  TowerForgeServiceUserAccessKey:
+    Type: AWS::IAM::AccessKey
+    Properties:
+      UserName: !Ref TowerForgeServiceUser
+
+  TowerForgeServiceRole:
+    Type: AWS::IAM::Role
+    Properties:
+      ManagedPolicyArns:
+        - !Ref TowerForgePolicyArn
+        - !Ref TowerLaunchPolicyArn
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              AWS:
+                - !GetAtt TowerForgeServiceUser.Arn
+            Action:
+              - sts:AssumeRole
+
+  EncryptionKeyStack:
+    Type: AWS::CloudFormation::Stack
+    Properties:
+      TemplateURL: !Sub ${TemplateRootUrl}/aws-infra/v0.3.1/KMS/kms-key.yaml
+      TimeoutInMinutes: 5
+      Parameters:
+        AliasName: !Sub "alias/${AWS::StackName}-EncryptionKey"
+        AdminPrincipalArns: !Join
+          - ","
+          - - !Sub "arn:aws:iam::${AWS::AccountId}:root"
+            - !Join
+              - ","
+              - !Ref AccountAdminArns
+        UserPrincipalArns: !Join
+          - ","
+          - - !Sub "arn:aws:iam::${AWS::AccountId}:root"
+            - !GetAtt TowerForgeServiceRole.Arn
+            - !Join
+              - ","
+              - !Ref AccountAdminArns
+            - !If
+              - HasS3ReadWriteAccessArns
+              - !Join
+                - ","
+                - !Ref S3ReadWriteAccessArns
+              - !Ref AWS::NoValue
+            - !If
+              - HasS3ReadOnlyAccessArns
+              - !Join
+                - ","
+                - !Ref S3ReadOnlyAccessArns
+              - !Ref AWS::NoValue
+
+  TowerBucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName: !Sub "${AWS::StackName}-tower-bucket"
+      OwnershipControls:
+        Rules:
+          - ObjectOwnership: BucketOwnerPreferred
+      # TODO: Re-enable encryption once Tower jobs have access to KMS key
+      # BucketEncryption:
+      #   ServerSideEncryptionConfiguration:
+      #     - BucketKeyEnabled: true
+      #       ServerSideEncryptionByDefault:
+      #         SSEAlgorithm: aws:kms
+      #         KMSMasterKeyID: !GetAtt EncryptionKeyStack.Outputs.KeyArn
+      LifecycleConfiguration:
+        Rules:
+        - Id: DataLifecycleRule
+          Status: !Ref EnableDataLifeCycle
+          ExpirationInDays: !Ref LifecycleDataExpiration
+          Transitions:
+            - TransitionInDays: !Ref LifecycleDataTransition
+              StorageClass: !Ref LifecycleDataStorageClass
+
+  TowerBucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref TowerBucket
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Sid: DenyAccessToNonAdmins
+            Effect: Deny
+            Principal:
+              AWS: !Ref AccountNonAdminArns
+            Action: "s3:*"
+            Resource:
+              - !Sub "arn:aws:s3:::${TowerBucket}"
+              - !Sub "arn:aws:s3:::${TowerBucket}/*"
```

Comment:
Instead of Deny, let's rewrite which permissions the Viewer role has. 
This only matters for Prod account b/c PHI will only be there. So Developer role doesn't matter -- it's not in Prod.

Reply:
The limited SSO Viewer role is implemented in Sage-Bionetworks-IT/organizations-infra#282.